_source/_posts/2022-07-11-kubernetes-jhipster-aws.md

```diff
@@ -0,0 +1,731 @@
+tags: [java,jhipster,kubernetes,k8s,devops,microservices]
+tweets:
+- "From soup 🍲 to nuts 🥜: here's everything you need to build a microservice stack and deploy it to Amazon Elastic Kubernetes Service #EKS. 
+Not-so-secret ingredients: @jhipster for your generator and @okta for #OIDC "
+- "Bootstrap complex apps with multiple microservices and domain models. @jhipster makes it easier. And Okta helps with #OAuth #security."
+- ""
+image:
+type: conversion
+
+---
+
+In this tutorial, you'll use JHipster to build a microservice stack and deploy it to Amazon Elastic Kubernetes Service (EKS). The microservices will use Java and Spring Boot for resource servers and Vue for the frontend. The server-side applications will use Spring's Reactive technology stack. The microservice will include multiple databases and database types: PostgresSQL, MongoDB, and Neo4j. You'll secure the service using OAuth 2.0 and OpenID Connect (OIDC), with Okta as the OIDC provider. With Kubernetes secrets and `kubeseal`, you'll encrypt all of the secrets in the project configuration files. To generate the project, you'll use JHipster CLI and the JHipster Domain Language (JDL).
+
+**Prerequisites**
+
+Before you get started, install the required software listed below. You'll need to sign up for an Amazon Web Services account and a free Okta account (which you will be able to sign up for later using the Okta CLI). The Kubernetes cluster required to finish this tutorial **does not qualify for free tier on AWS**. However, **if you are careful about stopping and deleting the cluster** when not actively working on the tutorial, the cost should be very small, only a few dollars.
+
+- [Docker](https://docs.docker.com/get-docker/): you’ll need to have **Docker Engine** installed.
+- [Docker Hub](https://hub.docker.com/): you’ll need a Docker Hub to host the Docker images so that EKS can pull them.
+- [Java 11](https://adoptopenjdk.net/): this tutorial requires Java 11. If you need to manage multiple Java versions, SDKMAN! is a good solution. Check out [their docs to install it](https://sdkman.io/install).
+- [Okta CLI](https://cli.okta.com/manual/#installation): you’ll use Okta to add security to the microservice network. You can register for a free account from the CLI.
+- [Amazon Web Services](https://aws.amazon.com/free): sign up for an AWS account.
+- [AWS CLI](https://aws.amazon.com/cli/): Amazon Web Service CLI
+- [eksctl](https://github.com/weaveworks/eksctl): you'll use `eksctl` to create your cluster on EKS
+- [kubectl](https://kubernetes.io/docs/tasks/tools/): CLI to manage Kubernetes clusters. As of the time I was writing this tutorial, there was a bug in `kubectl` version 1.24 that results in an "invalid apiVersion" error. You can avoid this error by installing version 1.23.6.
+- [JHipster CLI](https://www.jhipster.tech/installation/): you'll use the JHipster CLI to bootstrap the application. To install with `npm` just use `npm install -g generator-jhipster`
```

Comment:
nit: keep `kubectl` on the bottom so the follow-up on the error has more context

```suggestion
- [JHipster CLI](https://www.jhipster.tech/installation/): you'll use the JHipster CLI to bootstrap the application. To install with `npm` just use `npm install -g generator-jhipster`
- [Okta CLI](https://cli.okta.com/manual/#installation): you’ll use Okta to add security to the microservice network. You can register for a free account from the CLI.
- [Amazon Web Services](https://aws.amazon.com/free): sign up for an AWS account.
- [AWS CLI](https://aws.amazon.com/cli/): Amazon Web Service CLI
- [eksctl](https://github.com/weaveworks/eksctl): you'll use `eksctl` to create your cluster on EKS
- [kubectl](https://kubernetes.io/docs/tasks/tools/): CLI to manage Kubernetes clusters. As of the time I was writing this tutorial, there was a bug in `kubectl` version 1.24 that results in an "invalid apiVersion" error. You can avoid this error by installing version 1.23.6.
```